### PR TITLE
feat: block contenttype other than application/json

### PR DESCRIPTION
**Pull request by msohailhussain**
## Summary
-  Allow only application/json as content type for requests. Other content-type are not allowed.

## Test Plan
- Added unit tests

## Issues
OASIS-6582

### CODE REVIEW

**Comment by msohailhussain** [comment]
@mjc1283 I know it's not needed but what about if later on admin allows binary stuff?

**Comment by msohailhussain** [comment]
Based on our offlien discussion, we don't need configuration contentype, will remove that code.

File: pkg/routers/api.go [hunk 1]
```diff
@@ -124,6 +127,7 @@ func WithAPIRouter(opt *APIOptions, r chi.Router) {
 	}
 
 	r.Use(middleware.SetTime)
+	r.Use(opt.allowedContentTypesMiddleware)
```
Comment (mjc1283) [inline pkg/routers/api.go:130]: Why do we need to add `allowedContentTypesMiddleware` to `APIOptions`? Instead can we just do this here: 
```suggestion
	r.Use(chimw.AllowContentType("application/json"))
```

**Comment by mjc1283** [comment]
@yasirfolio3 You can apply the content type checking middleware only to routes that accept `POST` method.

**Pull request merged**